src/firmware/app/control/trajectory_planner.c

```diff
@@ -0,0 +1,356 @@
+#include "firmware/app/control/trajectory_planner.h"
+
+#include <assert.h>
+#include <stdbool.h>
+
+#include "assert.h"
+#include "firmware/shared/math/polynomial_1d.h"
+#include "firmware/shared/math/polynomial_2d.h"
+#include "math.h"
+
+TrajectoryPlannerGenerationStatus
+app_trajectory_planner_generateConstantArcLengthSegmentation(
+    FirmwareRobotPathParameters_t path_parameters,
+    Trajectory_t trajectory[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS])
+{
+    float max_allowable_speed_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    float velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+
+
+    // Set the internal path parameter variables
+    TrajectoryElement_t* traj_elements = trajectory->trajectory_elements;
+
+    float final_speed                      = path_parameters.final_speed;
+    float initial_speed                    = path_parameters.initial_speed;
+    const float max_allowable_acceleration = path_parameters.max_allowable_acceleration;
+    const float max_allowable_speed        = path_parameters.max_allowable_speed;
+    const unsigned int num_segments        = path_parameters.num_segments;
+    const Polynomial2dOrder3_t path        = path_parameters.path;
+    float t_end                            = path_parameters.t_end;
+    float t_start                          = path_parameters.t_start;
+
+    TrajectoryPlannerGenerationStatus generation_status = OK;
+
+
+    trajectory->num_elements = path_parameters.num_segments;
+
+    // Variable used to flag if the path is moving "backwards" along the input path
+    bool reverse_parameterization = false;
+
+    // Check that the pre conditions are met
+    assert(num_segments > 2);
```

Comment:
Probably should also check that `num_segments < TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS`

Reply:
Added assert and updated function @pre